dp-core/vr_interface.c

```diff
@@ -1860,15 +1862,50 @@ vr_interface_make_req(vr_interface_req *req, struct vr_interface *intf)
     req->vifr_obytes = 0;
     req->vifr_opackets = 0;
     req->vifr_oerrors = 0;
-
-    for (i = 0; i < vr_num_cpus; i++) {
-        stats = vif_get_stats(intf, i);
-        req->vifr_ibytes += stats->vis_ibytes;
-        req->vifr_ipackets += stats->vis_ipackets;
-        req->vifr_ierrors += stats->vis_ierrors;
-        req->vifr_obytes += stats->vis_obytes;
-        req->vifr_opackets += stats->vis_opackets;
-        req->vifr_oerrors += stats->vis_oerrors;
+    req->vifr_ifenqpkts = 0;
+    req->vifr_ifenqdrops = 0;
+    req->vifr_iftxrngenqpkts = 0;
+    req->vifr_iftxrngenqdrops = 0;
+    req->vifr_ifdeqpkts = 0;
+    req->vifr_ifdeqdrops = 0;
+    req->vifr_ifrxenqpkts = 0;
+    req->vifr_ifrxenqdrops = 0;
+
+    if (req->vifr_core == 0) {
+        for (i = 0; i < vr_num_cpus; i++) {
+            stats = vif_get_stats(intf, i);
+            req->vifr_ibytes += stats->vis_ibytes;
+            req->vifr_ipackets += stats->vis_ipackets;
+            req->vifr_ierrors += stats->vis_ierrors;
+            req->vifr_obytes += stats->vis_obytes;
+            req->vifr_opackets += stats->vis_opackets;
+            req->vifr_oerrors += stats->vis_oerrors;
+            req->vifr_ifenqpkts += stats->vis_ifenqpkts;
+            req->vifr_ifenqdrops += stats->vis_ifenqdrops;
+            req->vifr_iftxrngenqpkts += stats->vis_iftxrngenqpkts;
+            req->vifr_iftxrngenqdrops += stats->vis_iftxrngenqdrops;
+            req->vifr_ifdeqpkts += stats->vis_ifdeqpkts;
+            req->vifr_ifdeqdrops += stats->vis_ifdeqdrops;
+            req->vifr_ifrxenqpkts += stats->vis_ifrxenqpkts;
+            req->vifr_ifrxenqdrops += stats->vis_ifrxenqdrops;
+        }
+    } else if (req->vifr_core > 0) {
```

Comment:
We can get to vr_interface_make_req() either from vr_interface_get() or vr_interface_dump(). They both perform validation for vifr_core not to exceed maximum value. vr_interface_make_req() itself checks only if we want summed up stats or per-core. The vifr_core value is always sane there.